ratis-server/src/test/java/org/apache/ratis/server/impl/LeaderElectionTests.java

```diff
@@ -97,6 +100,70 @@ public void testBasicLeaderElection() throws Exception {
     cluster.shutdown();
   }
 
+  static class SleepCode implements CodeInjectionForTesting.Code {
+    private final long sleepMs;
+
+    SleepCode(long sleepMs) {
+      this.sleepMs = sleepMs;
+    }
+
+    @Override
+    public boolean execute(Object localId, Object remoteId, Object... args) {
+      try {
+        LOG.info("{}: Simulate RaftServer startup blocking", localId);
+        Thread.sleep(sleepMs);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+      return true;
+    }
+  }
+
+  @Test
+  public void testWaitServerReady() throws Exception {
+    LOG.info("Running testWaitServerReady");
+    int []arr = {1, 2, 3, 4, 5};
+    for (int i : arr) {
+      final MiniRaftCluster cluster = newCluster(1);
+      CodeInjectionForTesting.put(RaftServerImpl.START_COMPLETE, new SleepCode(1000));
```

Comment:
Let add a `CodeInjectionForTesting.remove` method.  Otherwise, it will affect other tests.

```diff
+++ b/ratis-common/src/main/java/org/apache/ratis/util/CodeInjectionForTesting.java
@@ -55,6 +55,12 @@ public final class CodeInjectionForTesting {
     INJECTION_POINTS.put(injectionPoint, code);
   }
 
+  /** Remove an injection point. */
+  public static void remove(String injectionPoint) {
+    final Code removed = INJECTION_POINTS.remove(injectionPoint);
+    LOG.debug("remove({}): {}", injectionPoint, removed);
+  }
+
   /** Execute the injected code, if there is any. */
   public static boolean execute(String injectionPoint, Object localId,
```